main.rkt

```diff
@@ -1,18 +1,28 @@
 #lang racket
 
 (require "enode.rkt" "egraph.rkt" "ematch.rkt" "extraction.rkt")
+(require debug/repl)
 
-(provide make-regraph regraph-cost regraph-count regraph-extract
-         rule-phase precompute-phase prune-phase extractor-phase)
 
-(struct regraph (egraph extractor ens limit))
+(module+ test (require rackunit math/base))
 
-(define (make-regraph exprs #:limit [limit #f])
+(provide make-regraph regraph-cost regraph-count regraph-extract regraph-limit
+         rule-phase precompute-phase prune-phase extractor-phase find-matches-time
+         regraph-match-count regraph-eclass-count rebuild-phase)
+
+(struct regraph (egraph extractor ens limit match-count rebuilding-enabled?) #:mutable)
+
```

Comment:
Probably would be best to replace `match-count` by some more general statistics object, which could also store the timing information.

Reply:
Haven't found a way to refactor `merge-time` out, but all the other timing is in a new rinfo struct.